src/modules/mavlink/mavlink_stream.cpp

```diff
@@ -114,11 +118,12 @@ MavlinkStream::update(const hrt_abstime t)
 		sent = send(t);
 #endif
 
-		// If the interval is non-zero do not use the actual time but
-		// increment at a fixed rate, so that processing delays do not
-		// distort the average rate
+		// If the interval is non-zero and dt is smaller than 1.5 times the interval
+		// do not use the actual time but increment at a fixed rate, so that processing delays do not
+		// distort the average rate. The check of the maximum interval is done to ensure that after a long time
+		// not sending anything multiple messages in a short time are sent.
```

Comment:
> The check of the maximum interval is done to ensure that after a long time not sending anything multiple messages in a short time are sent.

Should this not be more like: The check of the maximum interval is done to ensure that after a long time not sending anything, sending multiple messages in a short time is avoided.